code/datums/stock_market_events.dm

```diff
@@ -0,0 +1,115 @@
+/datum/stock_market_event
+	/// The name of the event, used to describe it in the news.
+	var/name = "Stock Market Event!"
+	/// A list of company names to use for the event and the circumstance.
+	var/static/list/company_name = list(
+		"Nakamura Engineering",
+		"Robust Industries, LLC",
+		"MODular Solutions",
+		"SolGov",
+		"Australicus Industrial Mining",
+		"Vey-Medical",
+		"Aussec Armory",
+		"Dreamland Robotics"
+	)
+	/// A list of strings selected from the event that's used to describe the event in the news.
+	var/list/circumstance = list()
+	/// What material is affected by the event?
+	var/datum/material/mat
+	/// Constant to multiply the original material value by to get the new minimum value, unless the material has a minimum override.
+	var/price_minimum = SHEET_MATERIAL_AMOUNT * 0.5
+	/// Constant to multiply the original material value by to get the new maximum value.
+	var/price_maximum = SHEET_MATERIAL_AMOUNT * 3
+
+	/// When this event is ongoing, what direction will the price trend in?
+	var/trend_value
+	/// When this event is triggered, for how long will it's effects last?
+	var/trend_duration
+
+/**
+ * When a new stock_market_event is created, this proc is called to set up the event if there's anything that needs to happen upon it starting.
+ * @param _mat The material that this event will affect.
+ */
+/datum/stock_market_event/proc/start_event(datum/material/_mat)
+	if(istype(_mat, /datum/material))
+		return FALSE
```

Comment:
To that end, if `mat` is NOT a reference, and is actually just a typepath, this is wrong and will always fail (should be using `ispath`)

Reply:
Yeah swapping to `ispath` breaks the code here